lib/foreman_rh_cloud/engine.rb

```diff
@@ -38,7 +38,7 @@ class Engine < ::Rails::Engine
 
         # Adding a sub menu after hosts menu
         sub_menu :top_menu, :foreman_rh_cloud, :caption => N_('RH Cloud'), :icon => 'fa fa-cloud-upload' do
-          menu :top_menu, :level1, :caption => N_('Inventory Upload'), :url_hash => { controller: :'foreman_inventory_upload/react', :action => :index}
+          menu :top_menu, :level1, :caption => N_('Inventory Upload'), :url_hash => { controller: :'foreman_rh_cloud/react', :action => :inventory_upload}
```

Comment:
Does this mean I will need to add an action to `react_controller.rb` for each menu item?

Reply:
basically what I did is in that action, I set the `@component_name` to render
if we can pass that as a `param` it can also work, can we?

Reply:
Yes, we can pass it as param.
@sharvit, is it a good practice?

Reply:
@sharvit some context,
We wanted to add react router but than we realized there is no registry yet for plugins routes, and we need to support foreman v1.20 too..

What we need is to handle several pages, should we mount each page component and use a sigle react controller?

Reply:
IMO it is too generic, having the `@component_name` and have it rendered by a generic erb layout.

I believe that a better approach would be to use a single controller for react (ReactController#index) but always render the same react-router layout component that would be aware of the route and would render the right component accordingly.

ReactController#index -> render react-router layout -> render the right page based on the location.

Reply:
Once https://github.com/theforeman/foreman/pull/6936 would merge, you would be able to delete some code and reuse the core logic.

Reply:
Thanks @sharvit !
That was the first thing that I did, though after realizing it's not going to be a real router
since every page will do a full page reload with the nav layout.

as we need to support also foreman 1.20 and above, I didn't find much value adding it as it will do the same as mounting each component.

another reason not to add to those version is that we need to make node packaging for `react-router-dom` for each version.

I do agree when https://github.com/theforeman/foreman/pull/6936 is merged we should use it and move only the last version to router which is already part of `@theforeman` packages which doesn't need packaging